text/0000-api-version-compat.md

```diff
@@ -0,0 +1,121 @@
+This RFC does not make changes to the API compatibility rules post 1.0. 1.x APIs version are stil assumed to be purely additive.
+
+## Lifecycle
+The lifecycle will use the [lifecycle descriptor](https://github.com/buildpacks/rfcs/blob/pack-publish-buildpack/text/0011-lifecycle-descriptor.md)
+to indicate the implemented API versions as before. However, the compatibility assumptions will change.
+It will be assumed that a lifecycle declaring support for the `0.x` version of the API, supports all 0.2-0.x API versions.
+
+### Platform API
+The lifecycle will use the `CNB_PLATFORM_API` environment variable to decide which API to implement. For now, if `CNB_PLATFORM_API` is not supplied, platform API 0.3 (the currently implemented API) will be assumed, to avoid breaking existing platforms that do not currently set this environment variable.
+
+If the API version declared in `CNB_PLATFORM_API` is higher than the newest supported by a given lifecycle, it will fail.
+
+### Buildpack API
+When orchestrating buildpacks, the lifecycle will provide inputs to and interpret outputs from each buildpack using the buildpack API indicated in that buildpack's `buildpack.toml` file.
+
+If the buildpack API declared in the buildpack.toml of a non-optional buildpack is higher than the newest supported by a given lifecycle, the lifecycle will fail.
+If the buildpack API declared in the buildpack.toml of an optional buildpack is higher than the newest supported by a given lifecycle, the lifecycle will warn, omit this buildpack from the group, and continue running.
+
+## Deprecations
+Once the 1.0 API is supported, we may wish to deprecate older APIs. A future RFC should provide a strategy for deprecations and outline the upgrade path from 0.x to 1.0 APIs. This is purposefully omitted from this RFC so that we can make progress on new 0.x APIs while we work out the details.
+
+# How it Works
+[how-it-works]: #how-it-works
+## Spec
+The core team must not allow changes in a 0.x API that necessarily prevents buildpacks implementing different APIs from running together.
+The contractual build plan is a danger zone for such changes. This does not necessarily mean no changes can be made to the contractual build plan,
+but any such changes should be made in close collaboration with the implementation team to ensure that the lifecycle can manage compatibility.
+
+## Lifecycle
+The lifecycle will maintain code for correctly implementing all supported APIs and choose which API to implement according to the rules above.
+
+## Platforms
+Platforms that wish to check lifecycle compatibility may now treat all minor API versions as non-breaking with the exception
+of buildpack and platform `0.1` -> `0.2`. Platforms should always set the `CNB_PLATFORM_API` environment variable to indicate
+which version of the API they are using.
+
+## Buildpacks
+Buildpacks should continue to declare the minimum required buildpack API in their `buildpack.toml`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
```

Comment:
I think another drawback is that it's difficult for a buildpack author interpret the cumulative set of changes between 0.2 and 0.x API versions.

Reply:
Added this drawback and a possible mitigation (upgrade docs)